frontend/src/store/settings.ts

```diff
@@ -83,7 +83,7 @@ export default function useSettingsStore() {
   }
 
   function setScanPrivateKey(key: string) {
-    if (key.length === 64) key = `0x${key}`;
+    if (key !== '' && key.length !== 66) throw new Error(`Invalid private key '${key}' ${key.length}`);
```

Comment:
If key length is 66, let's make sure `isHexString(key)` returns true (we should have already been doing this). Also I'd suggest tweaking the error message
```suggestion
    if (key !== '' && key.length !== 66) throw new Error(`Invalid private key '${key}'`);
```

Reply:
Added!

Reply:
Oh actually it seems like we already check for `isHexString` in `isValidPrivateKey`

Reply:
Guess it doesn't hurt to double check if we ever decide to use `setScanPrivateKey` elsewhere.

Reply:
Good point—it's probably redundant, but can't hurt in case the QInput validation fails / is accidentally broken in the future

I just pushed 2fd25a9 to fix the check / make it more readable (from my quick test the old version seemed incorrect). Give that a quick review/UI test, and if it works we can merge!

Reply:
Definitely more readable. The change is working as expected!